### PR TITLE
add compressed image messages

### PR DESCRIPTION
**Pull request by jveitchmichaelis**
Addresses https://github.com/Kukanani/vision_msgs/issues/20

### CODE REVIEW

**Comment by mintar** [comment]
The messages also need to be added to the CMakeLists.txt . Otherwise looks good. :)

**Comment by gavanderhoorn** [comment]
Not sure something can be done about it, but to me (non-native speaker) it sounds like the detections are compressed, not just the source image data.

**Comment by jveitchmichaelis** [comment]
As a native speaker I agree. The alternative is to roll it into the existing message, as the image is optional in any case? Then there's the issue of not knowing which image message to use. I don't have a strong opinion either way though.

**Comment by Kukanani** [comment]
Thanks for the PR!

I'd prefer multiple message types for clarity. As far as naming goes, I agree that `CompressedDetection` is slightly ambiguous. The only suggestion I have is something like `Detection2DWithCompression`, which is perhaps more explicit, but certainly more clunky. Instead of renaming the message, can you simply make a note in the header comments for the new message that the difference between this and the standard `Detection2D` is that the source image is compressed?

Also note #19, which I hope to merge tomorrow, would mean that we should include tracking information in this new message type as well.

**Comment by Kukanani** [comment]
I renamed the compression messages and added a comment in their headers about how they differ from the non-compression versions. Let me know if you feel the names are too clunky, otherwise I'll merge in a week or so.

**Comment by mintar** [comment]
Still missing the tracking fields.

**Comment by Kukanani** [comment]
Rebased and added tracking fields

File: msg/Detection2DWithCompressionArray.msg [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# A list of 2D detections, for a multi-object 2D detector.
```
Comment (jveitchmichaelis) [inline msg/Detection2DWithCompressionArray.msg]: Is it worth explicitly mentioning the compressed images in the header?

e.g. `A list of 2D detections with compressed images, for a multi-object 2D detector.`

**Review by jveitchmichaelis — COMMENTED** [review]
Thanks for tweaking this, apologies I've not had time to look at it. My only comment is maybe to add an explicit note in the detection array message that it's for compressed images?

**Comment by SteveMacenski** [comment]
Closing - we've removed the data sources entirely so this is no longer required

**Pull request closed without merging**